pkg/reconciler/v1alpha1/pipelinerun/pipelinerun_test.go

```diff
@@ -457,6 +470,172 @@ func TestUpdateTaskRunsState(t *testing.T) {
 
 }
 
+func TestUpdateTaskRunState_WithPassingConditionChecks(t *testing.T) {
+	pr := tb.PipelineRun("test-pipeline-run", "foo", tb.PipelineRunSpec("test-pipeline"))
+
+	cond := v1alpha1.Condition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "always-true",
+		},
+		Spec: v1alpha1.ConditionSpec{
+			Check: corev1.Container{},
+		},
+	}
+
+	taskCondition := v1alpha1.TaskCondition{
+		ConditionRef: "always-true",
+	}
+
+	pipelineTask := v1alpha1.PipelineTask{
+		Name:       "unit-test-1",
+		TaskRef:    v1alpha1.TaskRef{Name: "unit-test-task"},
+		Conditions: []v1alpha1.TaskCondition{taskCondition},
+	}
+
+	conditioncheck := conditionCheckFromTaskRun(tb.TaskRun("test-pipeline-run-success-unit-test-1-always-true", "foo", tb.TaskRunSpec(
+		tb.TaskRunTaskSpec(tb.TaskContainerTemplate()),
+	), tb.TaskRunStatus(
+		tb.Condition(apis.Condition{
+			Type:   apis.ConditionSucceeded,
+			Status: corev1.ConditionTrue,
+		}),
+		tb.StepState(tb.StateTerminated(0)),
+	)))
+
+	expectedConditionCheckStatus := make(map[string]*v1alpha1.PipelineRunConditionCheckStatus)
+	expectedConditionCheckStatus[conditioncheck.Name] = &v1alpha1.PipelineRunConditionCheckStatus{
+		ConditionName: cond.Name,
+		Status: &v1alpha1.ConditionCheckStatus{
+			Steps: []v1alpha1.StepState{{
+				ContainerState: corev1.ContainerState{
+					Terminated: &corev1.ContainerStateTerminated{ExitCode: 0},
+				},
+			}},
+			Status: duckv1beta1.Status{
+				Conditions: []apis.Condition{{Type: apis.ConditionSucceeded, Status: corev1.ConditionTrue}},
+			},
+		},
+	}
+
+	expectedTaskRunsStatus := make(map[string]*v1alpha1.PipelineRunTaskRunStatus)
+	expectedTaskRunsStatus["test-pipeline-run-success-unit-test-1"] = &v1alpha1.PipelineRunTaskRunStatus{
+		PipelineTaskName: "unit-test-1",
+		ConditionChecks:  expectedConditionCheckStatus,
+	}
+	expectedPipelineRunStatus := v1alpha1.PipelineRunStatus{
+		TaskRuns: expectedTaskRunsStatus,
+	}
+
+	state := []*resources.ResolvedPipelineRunTask{{
+		PipelineTask: &pipelineTask,
+		TaskRunName:  "test-pipeline-run-success-unit-test-1",
+		TaskRun:      nil,
+		ResolvedTaskResources: &taskrunresources.ResolvedTaskResources{
+			TaskSpec: &v1alpha1.TaskSpec{},
+		},
+		ResolvedConditionChecks: resources.TaskConditionCheckState{
+			{
+				ConditionCheckName: "test-pipeline-run-success-unit-test-1-always-true",
+				Condition:          &cond,
+				ConditionCheck:     conditioncheck,
+			},
+		},
+	}}
+	pr.Status.InitializeConditions()
+	updateTaskRunsStatus(pr, state)
+	if d := cmp.Diff(pr.Status.TaskRuns, expectedPipelineRunStatus.TaskRuns); d != "" {
+		t.Fatalf("Expected PipelineRun status to match ConditionCheck(s) status, but got a mismatch: %s", d)
+	}
+}
+
+func TestUpdateTaskRunState_WithFailingConditionChecks(t *testing.T) {
+	pr := tb.PipelineRun("test-pipeline-run", "foo", tb.PipelineRunSpec("test-pipeline"))
+
+	cond := v1alpha1.Condition{
```

Comment:
It would be good to have at least one of the two `updateTaskRunStatus` test to include the case of multiple conditions.

Reply:
Added a test case with multiple conditions